examples/nas/textnas/macro_child.py

```diff
@@ -0,0 +1,223 @@
+            self._linear_combine = LinearCombine(self.num_layers)
+        self.linear_out = nn.Linear(out_filters, self.class_num)
+
+        self.embed_dropout= nn.Dropout(p=(1 - embed_keep_prob))
+        self.output_dropout= nn.Dropout(p=(1 - final_output_keep_prob))
+
+        if self.output_type == "avg_pool":
+            self.output_pool = GlobalAvgPool()
+        elif self.output_type == "max_pool":
+            self.output_pool = GlobalMaxPool()
+        else:
+            raise ValueError("Unsupported output type.")
+
+    def forward(self, sent_ids, mask):
+        seq = Func.embedding(sent_ids.long(), self.embedding)
+        seq = self.embed_dropout(seq)
+
+        seq = torch.transpose(seq, 1, 2)  # from (N, L, C) -> (N, C, L)
+
+        x = self.init_conv(seq, mask)
+
+        start_idx = 0
+        prev_layers = []
+        final_flags = []
+
+        for layer_id in range(self.num_layers):  # run layers
+            layer = self.layers[layer_id]
+            x = self.run_fixed_layer(x, mask, prev_layers, layer, layer_id, start_idx,
+                                     final_flags=final_flags)  # run needed branches
+            prev_layers.append(x)
+            final_flags.append(1)
+
+            start_idx += 1 + layer_id
+            if self.multi_path:
+                start_idx += 1
+
+        final_layers = []
+        final_layers_idx = []
+        for i in range(0, len(prev_layers)):
+            if self.all_layer_output:
+                if self.num_last_layer_output == 0:
+                    final_layers.append(prev_layers[i])
+                    final_layers_idx.append(i)
+                elif i >= max((len(prev_layers) - self.num_last_layer_output), 0):
+                    final_layers.append(prev_layers[i])
+                    final_layers_idx.append(i)
+            else:
+                final_layers.append(final_flags[i] * prev_layers[i])
+
+        if self.all_layer_output and self.output_linear_combine:  # all layer ooutput and use linear_combine
+            x = self._linear_combine(torch.stack(final_layers))
+        else:
+            x = sum(final_layers)
+            if not self.all_layer_output:
+                x /= sum(final_flags)
+            else:
+                x /= len(final_layers)
+
+        x = self.output_pool(x, mask)
+        x = self.output_dropout(x)
+        x = self.linear_out(x)
+        return x
+
+    def make_fixed_layer(self, layer_id, out_filters):
+        size = [1, 3, 5, 7]
+        separables = [False, False, False, False]
+
+        branches = []
+
+        if self.multi_path:
+            branch_id = (layer_id + 1) * (layer_id + 2) // 2
+        else:
+            branch_id = (layer_id) * (layer_id + 1) // 2
+
+        bn_flag = False
+        for i in range(layer_id):
+            if self.sample_arc[branch_id + 1 + i] == 1:
+                bn_flag = True
+        branch_id = self.sample_arc[branch_id]
+
+        for operation_id in [0, 1, 2, 3]:  # conv_opt
```

Comment:
I notice that this is a LayerChoice but it's not using NNI interface. Theoretically, with NNI, you can share the same model class in search and retrain. Are you having any trouble with that?

Reply:
Totally agree, it is better to adopt the NNI interface. I need more time to learn the NNI interface and do the adaption.